src/Microsoft.AspNet.OData.Shared/Query/Expressions/ExpressionBinderBase.cs

```diff
@@ -1103,27 +1100,6 @@ internal static Expression ConvertNull(Expression expression, Type type)
             }
         }
 
-        /// <summary>
-        /// Compares two guids
-        /// </summary>
-        /// <param name="firstValue"></param>
-        /// <param name="secondValue"></param>
-        /// <returns>An integer value based on the Guid's CompareTo method</returns>
-        public static int GuidCompare(Guid firstValue, Guid secondValue)
-        {
-            if (firstValue != null)
-            {
-                return firstValue.CompareTo(secondValue);
-            }
-
-            if (secondValue != null)
-            {
-                return (-1) * secondValue.CompareTo(firstValue);
-            }
-
-            return 0;
```

Comment:
This is technically a breaking change to a public interface, although I doubt it is being used we could keep it here and just not use it when building the expression (and remove in 8.0).

I'm also fine removing if we are pretty confident it's not being used externally.

Reply:
I do think there's less customer who relies on this method.  It's "bad" method and we should fix before it broadcasts to more customer. Besides, it's a "easy" task at customer side to re-create a similar GUID compare method. So, in my opinion, let's remove it.